tests/test_processing_stack.py

```diff
@@ -263,31 +313,58 @@ def should_successfully_run_dataset_version_creation_process_with_multiple_asset
                 import_dataset_response, account_id, s3_control_client, subtests
             )
 
-            dataset_version_prefix = (
-                f"{dataset.title}{DATASET_KEY_SEPARATOR}{dataset.dataset_id}"
-                f"/{dataset_versions_body[VERSION_ID_KEY]}/"
-            )
+            dataset_version_prefix = f"{dataset_prefix}/{dataset_versions_body[VERSION_ID_KEY]}/"
             storage_bucket_prefix = f"{S3_URL_PREFIX}{Resource.STORAGE_BUCKET_NAME.resource_name}/"
 
+            with subtests.test(msg="Should update dataset catalog successfully"):
+                # Then poll dataset catalog for expected link to child catalog
+                expected_child_link_object = {
+                    STAC_HREF_KEY: f"./{dataset_versions_body[VERSION_ID_KEY]}"
+                    f"/{catalog_metadata_filename}",
+                    STAC_REL_KEY: STAC_REL_CHILD,
+                    STAC_TYPE_KEY: STAC_MEDIA_TYPE_JSON,
+                }
+                process_timeout = datetime.now() + timedelta(minutes=1)
+
+                while (
+                    expected_child_link_object
+                    not in (
+                        load(
+                            smart_open.open(
+                                f"{storage_bucket_prefix}{dataset_prefix}" f"/{CATALOG_FILENAME}",
+                                mode="rb",
+                            )
+                        )
+                    )[STAC_LINKS_KEY]
+                ):
+                    assert datetime.now() < process_timeout  # pragma: no cover
+                    sleep(5)  # pragma: no cover
+
             # Catalog contents
             imported_catalog_key = f"{dataset_version_prefix}{catalog_metadata_filename}"
             with subtests.test(msg="Imported catalog has relative keys"), smart_open.open(
                 f"{storage_bucket_prefix}{imported_catalog_key}", mode="rb"
             ) as imported_catalog_file:
                 assert load(imported_catalog_file) == {
                     **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                    STAC_EXTENSIONS_KEY: [],  # pystac writes this field.
                     STAC_LINKS_KEY: [
                         {
-                            STAC_HREF_KEY: collection_metadata_filename,
+                            STAC_HREF_KEY: f"./{collection_metadata_filename}",
```

Comment:
This is not a valid S3 URL/path - S3 doesn't have relative paths. So I'm not sure it'd be appropriate to have the "./" part at the start here, as anyone trying to process the dataset would probably have to do manual path munging to get the right path. Ditto for `../../` below.

Reply:
That depends on which layout strategy we intend to use.

In my mind we would be following the Relative Published strategy - where the root is an absolute s3 URL and everything else is a relative URL.

https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#relative-published-catalog

Reply:
I think that's a different thing. I'm just talking about how `./` and `../../` (as far as I know) won't do what you expect in an S3 link. For example, consider a user trying to crawl a dataset. They get to `s3://bucket/dataset/collection.json`, find a link `./foo.json`, and create a link to `foo.json` by concatenating `s3://bucket/dataset/` with `./foo.json`. I don't think that'll be a valid S3 link. They'll have to strip the `./` from the start before concatenating. Similarly, when dealing with links like `../catalog.json`, they'd have to strip a directory from the base URL for every `../` that they find in the link. On the other hand, since it seems S3 has no way to refer to a file in a parent directory with a relative link, we might as well pretend that we're on a "real" filesystem and use `./` and `../` for relative links.